frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx

```diff
@@ -85,9 +85,10 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({
           />
         </FormGroup>
       )}
-      {(noProjectsAvailable || selectedKey.value === CREATE_APPLICATION_KEY) && (
+      {(noProjectsAvailable || selectedKey.value !== UNASSIGNED_KEY) && (
```

Comment:
I think the issue with selecting an existing application and showing the Application Name field is tied to this.

Reply:
Maybe `noProjectsAvailable || noApplicationsAvailable || selectedKey.value === CREATE_APPLICATION_KEY?

Not sure why a project check is needed :/ I must have not noticed that during the other review when this went in. Can't have applications without projects. (this logic is mirrored in the above condition for the dropdown)

Could go with `noApplicationsAvailable || is CREATE_APPLICATION_KEY` - but I am weary about changing that logic without knowing more about it.